### PR TITLE
Various bug fixes in Eth API

### LINKED ISSUE #9636: /eth/v1/beacon/blocks returns incorrect error for missing block
# 🐞 Bug Report

### Description

Attempting to fetch a missing block using the standard API returns an error code 500, whereas it should return an error code 404.

### Has this worked before in a previous version?

No.

## 🔬 Minimal Reproduction

Against a prater node:

```
curl -v http://localhost:3501/eth/v1/beacon/blocks/4517
```

## 🔥 Error

<pre><code>
...
< HTTP/1.1 500 Internal Server Error
...
</code></pre>

As per https://ethereum.github.io/beacon-APIs/#/Beacon/getBlock this should return a 404.

## 🌍  Your Environment

develop.

### LINKED ISSUE #9647: /eth/v2/validator/blocks return wrong structure
# 🐞 Bug Report

### Description

The return block from an `eth/v2/validator/blocks` call is incorrectly wrapped in a `message` object.

## 🔬 Minimal Reproduction

```
curl -s 'http://localhost:3501/eth/v2/validator/blocks/1316809?randao_reveal=0x8d522fcf72b92b2f0164fa97d97c270dac746bb4db32481eaac5a114a66a130d93963285d7e3c5b7c7b16d10a5e362c20a21bcd82d509a4e6aa5a5a3b6f88b3db12fa67390a0b11ad1971b411906661e828924f024e67c1b360b99dcc442bdbd&graffiti=0x0000000000000000000000000000000000000000000000000000000000000000'
```

Change the slot number to the current slot for this to work.

## 🔥 Error

<pre><code>
{"version":"ALTAIR","data":{"message":{"slot":"1316809",...
</code></pre>

The return should read:

<pre><code>
{"version":"ALTAIR","data":{"slot":"1316809",...
</code></pre>

## 🌍  Your Environment

develop

### LINKED ISSUE #9652: Error requesting a ssz-encoded block from the API
# 🐞 Bug Report

### Description

When specifying content type via an Accept header as `application/octet-stream`, instead of receiving the requested block as an ssz-encoded value, the following error is observed. Note that w/o an accept header or by specifying `application/json`, a correct json encoded payload is returned. So the issue seems to be with the handling of requests for ssz encoded values.

### Has this worked before in a previous version?

Yes, this worked before I rebased my branch to pull in the changes to merge API serving on ports 3500/3501.

## 🔬 Minimal Reproduction

`curl -H "Accept: application/octet-stream" -i -X GET http://127.0.0.1:3500/eth/v2/beacon/blocks/0x6639eb3e2077f42fc8833ede7f46f96ee81027c0bc5e3907d1a72ce29a9ad7cd`

## 🔥 Error

<pre><code>:

{"message":"could not unmarshal error: invalid character 'p' after top-level value","code":500}

</code></pre>

### PR DESCRIPTION
**Pull request by rkapka**
**What type of PR is this?**

Bug fix

**What does this PR do? Why is it needed?**

- Fixes wrong block structure for `/eth/v2/validator/blocks`
- Changes status code to `NotFound` when block returned from core libraries is nil
- Make `/internal` grpc-gateway paths work with events and SSZ endpoints

**Which issues(s) does this PR fix?**

Fixes #9647
Fixes #9636
Fixes #9652

### CODE REVIEW

File: beacon-chain/rpc/apimiddleware/custom_hooks.go [hunk 1]
```diff
@@ -284,16 +284,12 @@ func serializeProducedV2Block(response interface{}) (bool, []byte, gateway.Error
 	if strings.EqualFold(respContainer.Version, strings.ToLower(ethpbv2.Version_PHASE0.String())) {
 		actualRespContainer = &phase0ProduceBlockResponseJson{
 			Version: respContainer.Version,
-			Data: &beaconBlockContainerJson{
-				Message: respContainer.Data.Phase0Block,
-			},
+			Data:    respContainer.Data.Phase0Block,
```
Comment (terencechain) [inline beacon-chain/rpc/apimiddleware/custom_hooks.go:287]: Does this need a regression test? Im surprised there's 0 test change to reflect on

File: beacon-chain/rpc/apimiddleware/custom_hooks.go [hunk 2]
```diff
@@ -265,13 +265,13 @@ func serializeV2State(response interface{}) (bool, []byte, gateway.ErrorJson) {
...
-	Data    *beaconBlockContainerJson `json:"data"`
+	Version string           `json:"version"`
+	Data    *beaconBlockJson `json:"data"`
 }
 
 type altairProduceBlockResponseJson struct {
-	Version string                          `json:"version"`
-	Data    *beaconBlockAltairContainerJson `json:"data"`
+	Version string                 `json:"version"`
+	Data    *beaconBlockAltairJson `json:"data"`
```
Comment (kasey) [inline beacon-chain/rpc/apimiddleware/custom_hooks.go:274]: presumably this is addressing `Fixes wrong block structure for /eth/v2/validator/blocks`?

Comment (rkapka) [reply beacon-chain/rpc/apimiddleware/custom_hooks.go:274]: Yes, this is correct. We don't need an additional wrapper, we should use `beaconBlockAltairJson` directly.

File: beacon-chain/rpc/apimiddleware/custom_hooks.go (hunk 1 above)
Comment (rkapka) [reply beacon-chain/rpc/apimiddleware/custom_hooks.go:287]: I improved block serialization tests.

**Review by terencechain — APPROVED** [review]

**Pull request merged**